src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -113,20 +120,14 @@ private unsafe void PrepareIOCPOperation()
             Debug.Assert(_currentSocket != null, "_currentSocket is null");
             Debug.Assert(_currentSocket.SafeHandle != null, "_currentSocket.SafeHandle is null");
             Debug.Assert(!_currentSocket.SafeHandle.IsInvalid, "_currentSocket.SafeHandle is invalid");
+            Debug.Assert(_preAllocatedOverlapped != null);
 
             ThreadPoolBoundHandle boundHandle = _currentSocket.GetOrAllocateThreadPoolBoundHandle();
 
             NativeOverlapped* overlapped = null;
-            if (_preAllocatedOverlapped != null)
-            {
-                overlapped = boundHandle.AllocateNativeOverlapped(_preAllocatedOverlapped);
-                if (NetEventSource.IsEnabled) NetEventSource.Info(this, $"boundHandle:{boundHandle}, PreAllocatedOverlapped:{_preAllocatedOverlapped}, Returned:{(IntPtr)overlapped}");
-            }
-            else
-            {
-                overlapped = boundHandle.AllocateNativeOverlapped(s_completionPortCallback, this, null);
-               if (NetEventSource.IsEnabled) NetEventSource.Info(this, $"boundHandle:{boundHandle}, AllocateNativeOverlapped(pinData=null), Returned:{(IntPtr)overlapped}");
-            }
+            overlapped = boundHandle.AllocateNativeOverlapped(_preAllocatedOverlapped);
```

Comment:
Nit: can be assigned directly to overlapped when it's declared on the previous line